### PR TITLE
Private contract

### PR DESCRIPTION
**Pull request by kirushik**
This PR also gets rid of `Multisig.sol` (as discussed with @kaeptnjoda, @jacogr and @lght), and revitalizes our Travis setup for contracts.

### CODE REVIEW

**Comment by arkpar** [comment]
Private contract related changes look good to me

File: contracts/PrivateContract.sol [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        code = initialCode;
+        state = initialState;
+        nonce = 1;
+    }
+
+    function setState(bytes newState, uint8[] v, bytes32[] r, bytes32[] s) public {
+        var noncedStateHash = keccak256([keccak256(newState), bytes32(nonce)]);
+
+        for (uint i = 0; i < validators.length; i++) {
+            assert(ecrecover(noncedStateHash, v[i], r[i], s[i]) == validators[i]);
```
Comment (rphmeier) [inline contracts/PrivateContract.sol]: requires unanimous approval? shouldn't a majority suffice?

**Comment by rphmeier** [comment]
#94 contains some manticore stuff too. can you explain the overlap/differences? are they mutually exclusive?

**Comment by kirushik** [comment]
@rphmeier About manticore stuff -- we've been just playing with the possibility to get manticore and Python in out CI pipeline right in this PR. Looks like it doesn't play out that well, so I've removed that commit from this PR, it's only Truffle for now.

**Comment by rphmeier** [comment]
Looks good to me as an initial version, although my concerns about the actual usability of unanimous signatures remain.

**Pull request merged**